### PR TITLE
Fix Honda checksum for extended CAN IDs. Add test function.

### PR DESCRIPTION
**Pull request by csouers**
Fixes a regression from https://github.com/commaai/opendbc/pull/568. `address` gets decimated so we can't use it for the extended id check. This also adds a test that checks both Honda checksum styles against a set of known good values.

Verified by Discord user: RedAcid
Vehicle: 2020 Honda Civic Hatchback

### CODE REVIEW

File: can/tests/test_packer_parser.py [hunk 1]
```diff
@@ -182,6 +182,43 @@ def test_updated(self):
       if len(user_brake_vals):
         self.assertEqual(vl_all[-1], parser.vl["VSA_STATUS"]["USER_BRAKE"])
 
+  def test_honda_checksum(self):
```
Comment (adeebshihadeh) [inline can/tests/test_packer_parser.py:185]: can you pull this into a separate file called test_checksums?

Comment (jyoung8607) [reply can/tests/test_packer_parser.py:185]: +1, been meaning to do this for VW. Will do so once a place/structure is settled on.

Comment (csouers) [reply can/tests/test_packer_parser.py]: Done. Comments?

**Comment by adeebshihadeh** [comment]
Thanks for the fix, and especially for writing tests :)

**Comment by csouers** [comment]
> Thanks for the fix, and especially for writing tests :)

Any time!

**Pull request merged**